scarb/src/ops/resolve.rs

```diff
@@ -222,6 +265,13 @@ fn generate_cairo_compilation_units(
         .collect())
 }
 
+fn is_integration_test(target: &Target, lib_target: Option<Target>) -> bool {
+    let lib_target_source_path = lib_target
+        .map(|target| target.source_path.to_string())
+        .unwrap_or(DEFAULT_SOURCE_PATH.to_string());
+    target.is_test() && !target.source_path.ends_with(lib_target_source_path)
```

Comment:
Not sure about this cond 🤔

Reply:
Either this stays, it's quite fine, or we store an enum in test target props which states whether this is unit test or integration test.